tools/gbagfx/rl.c

```diff
@@ -2,29 +2,38 @@
 
 #include <stdlib.h>
 #include <stdbool.h>
+#ifndef DO_UNSAFE_CONVERT
 #include "global.h"
 #include "rl.h"
+#endif
 
+#ifdef DO_UNSAFE_CONVERT
+void RLDecompress(unsigned char *src, unsigned char *dest, int *uncompressedSize)
+#else
 unsigned char *RLDecompress(unsigned char *src, int srcSize, int *uncompressedSize)
+#endif
 {
-    if (srcSize < 4)
-        goto fail;
-
-    int destSize = (src[3] << 16) | (src[2] << 8) | src[1];
+#ifndef DO_UNSAFE_CONVERT
```

Comment:
Yeah again with including these in the sa2 build if you are having to modify the functions then copying them would be fine

Reply:
I just don't see, why we should have multiple copies of it in the codebase. Just because #ifdefs are a little ugly?
Don't think that is enough reason.